lib/diaspora/webfinger.rb

```diff
@@ -0,0 +1,121 @@
+#   Copyright (c) 2010-2012, Diaspora Inc.  This file is
+#   licensed under the Affero General Public License version 3 or later.  See
+#   the COPYRIGHT file.
+
+module Diaspora
+  class Webfinger
+    include Diaspora::Logging
+
+    attr_accessor :host_meta_xrd, :webfinger_profile_xrd,
+                  :webfinger_profile, :hcard, :hcard_xrd, :person,
+                  :account, :ssl
+
+    def initialize(account)
+      self.account = account
+      self.ssl = true
+    end
+
+    def fetch
+      return person if existing_person_with_profile?
+      create_or_update_person_from_webfinger_profile!
+    end
+
+    def self.in_background(account, opts={})
```

Comment:
Unused method argument - `opts`. If it's necessary, use `_` or `_opts` as an argument name to indicate that it won't be used.